bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAuxiliaryHandler.java

```diff
@@ -0,0 +1,139 @@
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.openwebnet.internal.OpenWebNetBindingConstants;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openwebnet4j.OpenGateway;
+import org.openwebnet4j.communication.OWNException;
+import org.openwebnet4j.message.Auxiliary;
+import org.openwebnet4j.message.BaseOpenMessage;
+import org.openwebnet4j.message.MalformedFrameException;
+import org.openwebnet4j.message.OpenMessage;
+import org.openwebnet4j.message.UnsupportedFrameException;
+import org.openwebnet4j.message.Where;
+import org.openwebnet4j.message.WhereAuxiliary;
+import org.openwebnet4j.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetAuxiliaryHandler} is responsible for sending Auxiliary (AUX) commands/messages to the bus
+ * It extends the abstract {@link OpenWebNetThingHandler}.
+ *
+ * NOTICE: Support for handling messages from the bus regarding alarm control has to be implemented
+ *
+ * @author Giovanni Fabiani - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class OpenWebNetAuxiliaryHandler extends OpenWebNetThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetAuxiliaryHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.AUX_SUPPORTED_THING_TYPES;
+
+    public OpenWebNetAuxiliaryHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        OpenWebNetBridgeHandler h = bridgeHandler;
+        if (h != null && h.isBusGateway()) {
+            OpenGateway gw = h.gateway;
+            if (gw != null && gw.isConnected()) {
+                try {
+                    OpenMessage msg = BaseOpenMessage.parse("*#9##");
+                    send(msg);
+                } catch (MalformedFrameException | UnsupportedFrameException | OWNException e) {
+                    logger.warn("Exception while processing command {}: ", e.getMessage());
+                }
+            } else {
+                logger.warn("Gateway is not connected");
+            }
+        } else {
+            logger.warn("Gateway is not available");
+        }
+    }
+
+    /**
+     * Handles Auxiliary command for a channel
+     *
+     * @param channel the channel
+     * @param command the Command
+     */
+    @Override
+    protected void handleChannelCommand(ChannelUID channel, Command command) {
+        logger.debug("handleAuxiliaryCommand() (command={} - channel={})", command, channel);
+        Where w = deviceWhere;
+        if (w != null) {
+            if (channel.getId().equals(CHANNEL_AUX)) {
+                updateStatus(ThingStatus.ONLINE);
```

Comment:
This is not the receiving of a user command that should change the thing status.
Line to be removed.
Can you explain what logic you try to implement ?

Reply:
The thing has to change  its status to   ONLINE  if the right channel has been  grabbed and all variables have been  properly set!

Reply:
Here is the definition of a thing status: "Each Thing has a status object, which helps to identify possible problems with the device or service."
https://www.openhab.org/docs/concepts/things.html#thing-status

Just sending a command to a thing channel is not a correct trigger to change its status.

Generally, the thing status is updated when the communication with the device is broken. But when handling a user command, this is even not systematic, because you can consider that just a failign command sent to the device is not sufficient to consider that the communication with the device is fully broken.

So clearly, your change of status is not at the right place. You have the option to change the thing status to ONLINE or OFFLINE just few line below by handling the result of your sent command. Or you have the option to not update the thing status at all (in this case).

Reply:
actually as done in other handlers, thing should be set online when a status update is received from bus.
In this case is enough after sending of the command message.

Reply:
ok I see,  so the line 93 as been removed as @lolodomo and @mvalla  suggested. Thank you both  for the clarification!